### ISSUE TITLE
Doesn't recognise .Net 4.8 on Win 10

### ISSUE THREAD

**Issue by GLComputing** [MicrosoftDocs/OfficeDocs-Exchange#1619]
Trying to install the Exch 2019 CU4 Management Tools on Windows 10. 
It fails with an error that the system requires .Net 4.8 ... Which I have.
I even tried installing the .Net 4.8 for the link, which tells me that it's already installed on the computer.
And the links for more info go to pages on 2013.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 355e223c-c419-9fc8-16a9-0329e4794034
* Version Independent ID: 035ae216-9973-3804-df10-01f1f330e6c1
* Content: [Install the Exchange management tools](https://docs.microsoft.com/en-us/exchange/plan-and-deploy/post-installation-tasks/install-management-tools?view=exchserver-2019#feedback)
* Content Source: [Exchange/ExchangeServer/plan-and-deploy/post-installation-tasks/install-management-tools.md](https://github.com/MicrosoftDocs/OfficeDocs-Exchange/blob/public/Exchange/ExchangeServer/plan-and-deploy/post-installation-tasks/install-management-tools.md)
* Product: **exchange-server-it-pro**
* Technology: ****
* GitHub Login: @mattpennathe3rd
* Microsoft Alias: **v-mapenn**

**Comment by GLComputing** [comment]
![Exch-Net](https://user-images.githubusercontent.com/35485784/71545351-3eb8d300-29de-11ea-861b-a7b23e92e96c.jpg)

**Comment by keithw89** [comment]
I am having the same issue, when will this be fixed?

**Comment by GiantCrocodile** [comment]
This is most likely a product issue then. Did you raise a support ticket?

**Comment by JeremyTBradshaw** [comment]
While I won't open the support ticket for this, I will add that I too have stumbled upon this same issue.  Following the article here - https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed - I can see in my registry that I have "Version": 4.8.03752.  But I get the same error message from the Exchange installer, and from the .Net 4.8 installer as @GLComputing shared the screenshots for.

If you click the "?" button in the Exchange installer, you get taken to this decommissioned page for Exchange 2013 setup help: https://docs.microsoft.com/en-us/previous-versions/exchange-server/exchange-150/jj126480(v=exchg.150)?redirectedfrom=MSDN

That last point just to share.  I also emailed the support address from that page to point this out to them/there as well.

Back to opening a support ticket.  Now that I can see others have the issue, and we're not in the same org's, I'm hesitant to burn the cycles vs just report it here.  MS can choose to initiate effort from here or not.  I'm just going to hold off on putting the mgmt tools on Windows 10 for now.

**Comment by keithw89** [comment]
No I did not, can you provide an email address to open a support ticket?

**Comment by GiantCrocodile** [comment]
I understand your hesitation to open a support ticket regarding this @JeremyTBradshaw. You might be lucky that Microsoft's team here will communicate this properly and find a solution. Sometimes it does happen, sometimes it doesn't.

@keithw89 You have to be an user with admin privileges or other support/admin roles in your Office 365 tenant and then you can open a support ticket via the admincp at admin.microsoft.com or via your Cloud Service Provider (CSP) in case you don't buy licenses at Microsoft directly. If you don't have those permissions, there's no way to do it. You might have to ask your IT department then.

**Comment by mattpennathe3rd** [comment]
@GLComputing @keithw89 @GiantCrocodile @JeremyTBradshaw Everyone, I'm sorry, but I am unable to provide product support here. This repository is only for reporting issues with the documentation and I'm not equipped to help with product issues. As suggested, please open a support ticket.

**Comment by GLComputing** [comment]
Tried to log a support ticket, but they want to charge over $400 before they'll even talk to me.

Is there a better way to log a bug?

**Comment by JeremyTBradshaw** [comment]
I do sympathize that this might not be the correct place to submit product bugs as it's just open documentation, not open-source software.

But, uservoice doesn't fit here and support ticket means the customer has to endure the ticket while Microsoft look into the bug, charging the customer in the process.  But the customer already paid and is just letting them know about a broken thing with the product.  I guess it's like any big company - some things will just never be possible like it would for a small company.

If there could just be a bug report email address or something where people could send the low hanging fruit issues like this one and MS could parse through and fix a few easy ones - that'd be great.

**Comment by GiantCrocodile** [comment]
@GLComputing How did you do that? Office 365 support is free. I think you tried a Premier support ticket which costs way more (400 or 500 $). If you don't see the support ticket option to contact Microsoft in your Office 365 tenant, you might use a CSP. Then your CSP has to deal with this and forward it to Microsoft. They are your first level support and responsible for escalating the ticket to Microsoft.

You don't have to pay per support ticket for Office 365 @JeremyTBradshaw.

**Comment by GiantCrocodile** [comment]
Nevermind, just deleted my previous answer as I did realize this is not related to Office 365 but an on-premises product so my previous statement about support via Office 365 admincp tenant is obviously wrong.

**Comment by JeremyTBradshaw** [comment]
@GiantCrocodile I have a falsity to report of my own.  Just realized this issue is for Exchange 2019.  I noticed the issue with Exchange 2016 CU15.
So I have no business stating anything here, but at this point I'll leave my posts just to preserve the order of the conversation.  It's really not a huge concern, there are definitely bigger issues to worry about.  I'll sign off now:)

**Comment by JulioQc** [comment]
Replicated issue on Windows 10 1909 trying to upgrade Exchange 2016 Management Tools from CU14 to CU15... Even uninstall / install comes up with the error: 

`This computer requires .NET Framework 4.8 (https://support.microsoft.com/kb/4503548).
For more information, visit: https://docs.microsoft.com/Exchange/plan-and-deploy/system-requirements?view=exchserver-2016`

But obviously it was installed (actually for installing CU14 two months ago): 

`PS C:\WINDOWS\system32> (Get-ItemProperty "HKLM:SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full").Release           `
`528040`

From this: https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed
I highly suspect their code looks for value 528049 but some might be 528040 on certain windows version. That value cannot easily be changed. 

Each upgrade is problematic. I wonder if this will ever go smoothly. Probably not until Microsoft teams communicate between them and implicate QA before release * sigh *

**Comment by andrewfake** [comment]
Why is this comment closed? What is the solution? It constantly thinks .NET 4.8 is missing when it clearly is installed.

Someone help?

**Comment by JulioQc** [comment]
> Why is this comment closed? What is the solution? It constantly thinks .NET 4.8 is missing when it clearly is installed.
> 
> Someone help?

Because they probably will not fix anything and address the problem in CU16, which will likely introduce another issue. That's from the indian guy I got from support so I'll take it for truth. 

The update from CU14 to CU15 worked fine on the server 2016 but managing from Win10 is still being done from a CU14 install due to this error. I would recommend this as a quick fix. No major issue experienced so far managing CU15 from a CU14 (although this might not be supported, but again they support what they want to support when it suits them best).

**Comment by GLComputing** [comment]
I January, I spoke to someone at MS who got me to test a workaround for this and, after I verified it, posted it on this KB:
https://support.microsoft.com/en-au/help/4538303/exchange-2019-setup-prerequisite-check-fails-for-net-4-8-framework

While I only tested on Ex2019, I guess (hope) it might help those with the same issue on the latest 2016CU as well.

I thought I'd add it here for anyone who hits this page looking for a solution.

**Comment by billeuze** [comment]
> In January, I spoke to someone at MS who got me to test a workaround for this and, after I verified it, posted it on this KB:
> https://support.microsoft.com/en-au/help/4538303/exchange-2019-setup-prerequisite-check-fails-for-net-4-8-framework
> 
> While I only tested on Ex2019, I guess (hope) it might help those with the same issue on the latest 2016CU as well.
> 
> I thought I'd add it here for anyone who hits this page looking for a solution.

Confirmed, this works for Exchange 2016cu18 and win10-1909. Thanks for posting this workaraound

**Comment by cosmoKenney** [comment]
I just see a spinner when I go to this page: https://support.microsoft.com/en-au/help/4538303/exchange-2019-setup-prerequisite-check-fails-for-net-4-8-framework

**Comment by billeuze** [comment]
> I just see a spinner when I go to this page: https://support.microsoft.com/en-au/help/4538303/exchange-2019-setup-prerequisite-check-fails-for-net-4-8-framework

It still works for me. Anyway If anyone cannot access the page for some reason here is the workaround posted on that page:

To work around this issue, follow these steps:

1. Start “regedit.exe” as an administrator.
2. Locate the following registry subkey:

 
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full

3. Right-click the subkey, and select Permissions from the shortcut menu.
4. Select Advanced.
5. In the Advanced Security Settings window, locate the Owner attribute at the top of the window.
6. Select Change next to the listed owner.
7. In the Enter the object name to select field, enter the name of the local administrator group. For example, enter <computername>\Administrator. Then, select OK.

8. In the Advanced Security Settings window, select the local administrator group that you changed ownership to, and then select Edit.
9. Change the basic permissions to Full Control.
10. Select OK three times to save the changes and return to the main Registry Editor window.
11. Locate the following key in the path:

 
Name: Release
Type: REG_DWORD
Data: 528040 (decimal)

Change the Data value to 528049 (decimal).

12. Rerun the Exchange system prerequisites check, and deploy or update Exchange Server 2019.
13. Start Registry Editor, locate the subkey that's mentioned in step 2, repeat the necessary steps to locate the Release key, and then revert the Data value to 528040 (decimal).

**Comment by kssonu1811** [comment]
> I am having the same issue, when will this be fixed?

i am also facing same problem
